config/user-libudev.m4

```diff
@@ -0,0 +1,13 @@
+dnl #
+dnl # Check for libudev - needed for vdev auto-online and auto-replace
+dnl #
+AC_DEFUN([ZFS_AC_CONFIG_USER_LIBUDEV], [
+	LIBUDEV=
+
+	AC_CHECK_HEADER([libudev.h], [AC_SUBST([LIBUDEV], ["-ludev"])
+	AC_DEFINE([HAVE_LIBUDEV], 1, [Define if you have libudev])], [])
```

Comment:
Please fix the indentation on these two checks as well.  The first _two_ times I read this it wasn't clear to me the AC_DEFAULT line was inside AC_CHECK_HEADER.  Same for AC_CHECK_LIB below.  Use the normal cstyle indentation rules here as much as possible.